08_education/college_metrics_county.qmd

```diff
@@ -0,0 +1,1171 @@
+```{r}
+acs <- extract_ipums(
+  extract_name = "umf_data_18_22",
+  extract_description = "Microdata pull for College Readiness Metric Predictors. American Community Survey, years 2018, 2019 and 2022 (5-year).",
+  survey = list("us2018c", "us2021c"))
+```
+
+Look at the distribution of ACS survey samples in the data. The number of different samples should align with the number of surveys selected in the extract ipums function.
+
+```{r}
+count(acs, sample)
+```
+
+Isolate the data to 19 and 20 year-olds. This is the age-range included in the college readiness calculation.
+```{r}
+acs <- acs %>%
+  filter(age == 19 | age == 20) 
+```
+
+
+### Replicate Weights 
+
+Read in person-level replicate weights for the 19 and 20 year old population. These will be used in creating standard errors for the college ready calculation. 
+```{r}
+repwts_person <- ipums_repwt_college(
+  extract_name = "college_replicate_weights",
+  extract_description = "Person replicate weights for College Readiness Metric Predictors. American Community Survey, years 2018 and 2022 (5-year).",
+  survey = list("us2018c", "us2021c")
+)  %>% 
+  select(-cbserial, -cbpernum)
+```
+
+Look at the distribution of ACS survey samples in the data. The number of different samples should align with the number of surveys selected in the extract ipums function.
+
+```{r}
+count(repwts_person, sample)
+```
+
+Remove the sample variable.
+```{r}
+repwts_person <- repwts_person %>% 
+  select(-sample)
+```
+
+### Merge on repwts
+
+```{r}
+length(unique(pull(acs, unique_person_id)))
+length(unique(pull(repwts_person, unique_person_id)))
+acs_combined <- 
+  left_join(
+    acs,
+    repwts_person,
+    by = "unique_person_id"
+  )
+length(unique(pull(acs_combined, unique_person_id)))
+rm(acs)
+rm(repwts_person)
+```
+
+Check that the merge did not result in any missing values.
+```{r}
+stopifnot(all(map_dbl(acs_combined, ~sum(is.na(.x))) == 0))
+```
+
+
+## Clean Data
+
+Missing data is reported in a variety of ways by IPUMS data. This step walks through the missing values in key variables and checks that we are dealing with them appropriately. 
+
+Age: No missing values reported in variable description. Check the distribution to confirm there are no abnormalities: the distribution of age looks reasonable (no bunching or extreme outliers).
+
+```{r}
+acs_combined %>% 
+  ggplot(aes(x = age)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Age Variable Results") +
+  ylab("Density")
+```
+
+EDUCD: N/As reported as "1" in all modern surveys. N/As were "000" in the 1980 and 1950 decennial census surveys (not used here).
+Variable codes explained [here]([https://usa.ipums.org/usa-action/variables/EDUC#codes_section) & see "Detailed codes" at bottom).
+Vacancy is reported as "0". 
+
+There is a significant share of records reported as "1" for the education variable. 
```

Comment:
I don't think this is true anymore given changes to survey samples and earlier subsetting to relevant ages.